tests/unit/lms/services/launch_verifier_test.py

```diff
@@ -169,48 +173,71 @@ def launch_verifier(self, pyramid_request):
 
     @pytest.fixture
     def pyramid_request(self, pyramid_request):
-        # `pyramid.testing.DummyRequest` sets `url` and `path_url` attrs without
-        # a path by default. oauthlib however normalizes the URL to include a
-        # path when it generates the signature base string, whereas PyLTI does
-        # not. Add an empty path to both to match a real application.
-        pyramid_request.url += "/"
-        pyramid_request.path_url += "/"
-
-        # Add the OAuth 1 params (version, nonce, timestamp, ...)
-        oauthlib_client = oauthlib.oauth1.Client(
-            pyramid_request.params["oauth_consumer_key"]
-        )
-        oauthlib_request = oauthlib.common.Request(
-            pyramid_request.url, pyramid_request.method
-        )
-        pyramid_request.params = dict(
-            oauthlib_client.get_oauth_params(oauthlib_request)
-        )
+        def set_url_from_params():
+            parsed_url = urlparse(pyramid_request.url)
+            parsed_url = parsed_url._replace(query=urlencode(pyramid_request.params))
+            pyramid_request.url = parsed_url.geturl()
 
-        sign(pyramid_request)
+        pyramid_request.set_url_from_params = set_url_from_params
 
         return pyramid_request
 
+    @pytest.fixture(autouse=True)
+    def oauth1(self, pyramid_request):
+        oauth1 = OAuth1Assistant(pyramid_request)
+        oauth1.add_oauth_params()  # Add the OAuth 1 params (version, nonce, timestamp, ...)
+        oauth1.sign()  # Sign the request
+
+        return oauth1
+
     @pytest.fixture
-    def pylti(self, patch):
-        pylti = patch("lms.services.launch_verifier.pylti")
-        pylti.common.LTIException = LTIException
-        return pylti
+    def oauth1_endpoint(self, launch_verifier):
+        oauth1_endpoint = create_autospec(
+            SignatureOnlyEndpoint, instance=True, spec_set=True
+        )
+        oauth1_endpoint.validate_request.return_value = True, sentinel.request
 
+        # pylint: disable=protected-access
+        launch_verifier._oauth1_endpoint = oauth1_endpoint
 
-def sign(pyramid_request):
-    """Add or replace pyramid_request's OAuth 1 signature param."""
-    oauthlib_client = oauthlib.oauth1.Client(
-        pyramid_request.params["oauth_consumer_key"], "TEST_SECRET"
-    )
-    oauthlib_request = oauthlib.common.Request(
-        pyramid_request.url, pyramid_request.method, body=pyramid_request.params
-    )
```

Comment:
:point_up: This right here is some odd behavior. The method in this case is `GET` we then take the params and pass them as a body, which `oauthlib` interprets as form fields and signs them.

It just so happens the way you sign the form fields and get params is identical, so it all sort of works out...

However, our new method requires the params to actually be on the URL we received _or_ for the request to be a proper POST with form fields.